orchestration/dagster_orchestration/hca_manage/validation.py

```diff
@@ -0,0 +1,90 @@
+from google.cloud import storage
+import logging
+from jsonschema import validate
+import json
+import requests
+from typing import Optional
+from hca_manage.common import DefaultHelpParser
+from urllib.parse import urlparse
+from dataclasses import dataclass
+
+
+class SchemaFetcher:
+    def __init__(self) -> None:
+        self._schema_cache = {}
+
+    def fetch_schema(self, path: str) -> Dict:
+        if path in self._schema_cache:
+            return self._schema_cache[path]
+
+        raw_schema = requests.get(path).json()
+        self._schema_cache[path] = raw_schema
+        return raw_schema
+
+
+@dataclass
+class GsBucketWithPrefix:
+    bucket: str
+    prefix: str
+
+
+def parse_gs_path(raw_gs_path: str) -> GsBucketWithPrefix:
+    url_result = urlparse(raw_gs_path)
+    return GsBucketWithPrefix(url_result.netloc, url_result.path[1:])
+
+
+def validate_json(blob: storage.Blob, schema_fetcher: Type[SchemaFetcher]) -> Optional[Exception]:
```

Comment:
⛏️ This is just a `SchemaFetcher`, not a `Type[SchemaFetcher]`
```suggestion
def validate_json(blob: storage.Blob, schema_fetcher:SchemaFetcher) -> Optional[Exception]:
```